File: samples/java-play2/app/Global.java
import play.*;

import com.wordnik.swagger.model.*;
import com.wordnik.swagger.config.*;
import com.wordnik.swagger.config.FilterFactory;

public class Global extends GlobalSettings {
  @Override
  public void onStart(Application app) {
    // do any additional initialization here, such as set your base path programmatically as such:
    // ConfigFactory.config().setBasePath("http://www.foo.com/");

    ApiInfo info = new ApiInfo(
      "Swagger Sample App",                             /* title */
      "This is a sample server Petstore server.  You can find out more about Swagger " + 
      "at <a href=\"http://swagger.wordnik.com\">http://swagger.wordnik.com</a> or on irc.freenode.net, #swagger.  For this sample, " + 
      "you can use the api key \"special-key\" to test the authorization filters", 
      "http://helloreverb.com/terms/",                  /* TOS URL */
      "apiteam@wordnik.com",                            /* Contact */
      "Apache 2.0",                                     /* license */
      "http://www.apache.org/licenses/LICENSE-2.0.html" /* license URL */
    );
    ConfigFactory.config().setApiInfo(info);
  }  
  
  @Override
  public void onStop(Application app) {
    Logger.info("Application shutdown...");
  }   
}

File: samples/scala-play2/app/Global.scala
import play.api._

import com.wordnik.swagger.config._
import com.wordnik.swagger.model._


object Global extends GlobalSettings {
  val info = ApiInfo(
    title = "Swagger Sample App",
    description = """This is a sample server Petstore server.  You can find out more about Swagger 
    at <a href="http://swagger.io">http://swagger.io</a> or on irc.freenode.net, #swagger.  For this sample,
    you can use the api key "special-key" to test the authorization filters""", 
    termsOfServiceUrl = "http://helloreverb.com/terms/",
    contact = "apiteam@wordnik.com", 
    license = "Apache 2.0", 
    licenseUrl = "http://www.apache.org/licenses/LICENSE-2.0.html")

  val oauth = OAuth(
    List(
      AuthorizationScope("write:pets", "Modify pets in your account"),
      AuthorizationScope("read:pets", "Read your pets")),
    List(
      ImplicitGrant(
        LoginEndpoint("http://petstore.swagger.wordnik.com/oauth/dialog"),
        "access_token"
      ),
      AuthorizationCodeGrant(
        TokenRequestEndpoint("http://petstore.swagger.wordnik.com/oauth/requestToken",
          "client_id",
          "client_secret"),
        TokenEndpoint("http://petstore.swagger.wordnik.com/oauth/token",
          "auth_code"
        )
    )
  ))
  ConfigFactory.config.authorizations = List(oauth)
  ConfigFactory.config.info = Some(info)
}